### PR TITLE
mm/map/vm_region.c: Fix usage of void* arithmetics

### PR DESCRIPTION
**Pull request by pussuw**
## Summary
Fixes build error:
map/vm_region.c: In function 'vm_map_region':
map/vm_region.c:128:16: error: pointer of type 'void *' used in arithmetic [-Werror=pointer-arith]
  128 |   return vaddr + (MM_PGMASK & paddr);
      |                ^
cc1: all warnings being treated as errors

## Impact
Fix build error
## Testing
Build / CI pass

### CODE REVIEW

File: mm/map/vm_region.c [hunk 1]
```diff
@@ -125,7 +125,7 @@ FAR void *vm_map_region(uintptr_t paddr, size_t size)
         }
     }
 
-  return vaddr + (MM_PGMASK & paddr);
+  return (void *)((uintptr_t)vaddr + (MM_PGMASK & paddr));
 
 error:
   if (i)   /* undo alway mapped pages */
```
Comment (acassis) [inline mm/map/vm_region.c]: Please fix the original issue that was in the file moving the comment to inside the: if (i) { /* here */ ... }

**Review by acassis — APPROVED** [review]

File: mm/map/vm_region.c [hunk 2]
```diff
@@ -125,7 +125,7 @@ FAR void *vm_map_region(uintptr_t paddr, size_t size)
         }
     }
 
-  return vaddr + (MM_PGMASK & paddr);
+  return (void *)((uintptr_t)vaddr + (MM_PGMASK & paddr));
```
Comment (xiaoxiang781216) [inline mm/map/vm_region.c:128]: add FAR

File: mm/map/vm_region.c [hunk 3]
```diff
@@ -128,8 +128,10 @@ FAR void *vm_map_region(uintptr_t paddr, size_t size)
   return (void *)((uintptr_t)vaddr + (MM_PGMASK & paddr));
 
 error:
-  if (i)   /* undo alway mapped pages */
+  if (i)
     {
+      /* undo alway mapped pages */
```
Comment (xiaoxiang781216) [inline mm/map/vm_region.c:133]: Undo

File: mm/map/vm_region.c (hunk 1 above)
Comment (pussuw) [reply mm/map/vm_region.c]: I did that

File: mm/map/vm_region.c (hunk 3 above)
Comment (pussuw) [reply mm/map/vm_region.c:133]: That produces an error from the format check `mm/map/vm_region.c:131:7: error: If statement followed by garbage`

Comment (acassis) [reply mm/map/vm_region.c:133]: Maybe you can put it after up_shmdt(); or maybe we need to modify nxstyle.c to fix this false positive

Comment (pussuw) [reply mm/map/vm_region.c:133]: Ah, you simply meant to capitalize the comment :)

I'll do that of course

Comment (xiaoxiang781216) [reply mm/map/vm_region.c:133]: Yes, sorry for confusion:).

**Pull request merged**